arduino/health_monitor/health_monitor.ino

```diff
@@ -327,54 +233,77 @@ void show_treatment_animation() {
 }
 
 void setup() {
-    configure_hw();
+    configure_hw(g_player_state_persistent);
 
-    Serial.println("Start Health: " + String(g_health_state.health));
+    Serial.println("Start Health: " + String(g_player_state_persistent.health.health));
 
     if (!is_monitor_enabled()) {
         Serial.println("Health Monitor disabled");
-        reset_state();
-    } else if (g_treated) {
-        Serial.println("Treatment received");
-        show_treatment_animation();
-        g_treated = false;
-
-        // apply treatment to the game state
-        g_health_state = apply_treatment(g_health_state);
+        game_reset(g_player_state_persistent);
     } else {
-        print_wakeup_reason();
-        // Increment boot number and print it every reboot
-        ++g_boot_count;
-        Serial.println("Boot number: " + String(g_boot_count));
-
         // String representing our state (used by other devices to observe us)
-        auto const display_state = to_display_state(g_health_state.health);
+        auto const display_state = to_display_state(g_player_state_persistent.health.health);
         Serial.println(display_state.c_str());
-        Serial.println("Health: " + String(g_health_state.health));
-        Serial.println("Cat Resistance: " + String(g_health_state.cat_resistance));
+        Serial.println("Health: " + String(g_player_state_persistent.health.health));
+        Serial.println("Cat Resistance: " +
+                       String(g_player_state_persistent.health.cat_resistance));
 
         // Start bluetooth to advertise our state
         BLEDevice::init(PREFIX_STR + display_state);
         BLEDevice::startAdvertising();
 
-        // Scan for nearby devices and count infected
-        auto devices = scan_ble();
-        print_scan_results(devices);
-        auto const exposure = count_exposure(devices);
-        Serial.println("Infected human exposure: " + String(exposure.human));
-        Serial.println("Infected cat exposure: " + String(exposure.cat));
-
-        // Update our health value
-        g_health_state = health_update(g_health_state, exposure);
+        // Event queue will hold semi-dynamic events to process on this tick
+        std::vector<Event> event_queue;
+        event_queue.reserve(3UL);
```

Comment:
This doesn't need to be a `vector`... there's a finite number of things that can happen in an updater ... could be a statically sized buffer

Reply:
:+1 for making statically sized queues.

Reply:
Will do in follow-up with a quick util class.